pool.go

```diff
@@ -292,213 +326,276 @@ func (ctx *valueCancelCtx) Value(key any) any           { return ctx.valueCtx.Va
 // ctx. Canceling ctx will cause Acquire to return immediately but it will not cancel the resource creation. This avoids
 // the problem of it being impossible to create resources when the time to create a resource is greater than any one
 // caller of Acquire is willing to wait.
-func (p *Pool[T]) Acquire(ctx context.Context) (*Resource[T], error) {
-	startNano := nanotime()
-	if doneChan := ctx.Done(); doneChan != nil {
-		select {
-		case <-ctx.Done():
-			p.cond.L.Lock()
-			p.canceledAcquireCount += 1
-			p.cond.L.Unlock()
-			return nil, ctx.Err()
-		default:
-		}
+func (p *Pool[T]) Acquire(ctx context.Context) (_ *Resource[T], err error) {
+	select {
+	case <-ctx.Done():
+		p.canceledAcquireCount.Add(1)
+		return nil, ctx.Err()
+	default:
 	}
 
-	p.cond.L.Lock()
+	return p.acquire(ctx)
+}
 
-	emptyAcquire := false
+// acquire is a continuation of Acquire function that doesn't check context
+// validity.
+//
+// This function exists solely only for benchmarking purposes.
```

Comment:
Thanks, I haven't known of this feature and the links you have shared are great! :)

Unfortunately because the `Acquire` function contains `select` statement, it cannot be inlined (it's mentioned as example of "hairy" statement in the first link). I tested this fact by writing sample program calling `Acquire` and it's really not inlined :/

On the other hand given that cost of `Acquire` call is already non-trivial even in case of idle resource acquire (remember semaphore call won't be inlined and requires allocation in case of contention), it doesn't feel to be big issue in this case. 

But it's really good to know about this inlining phenomenon, so thanks for sharing this knowledge once again :)